tests/expose_test.go

```diff
@@ -474,8 +797,9 @@ var _ = Describe("[rfe_id:253][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 				Eventually(endptSubsetIpAddress).Should(BeEquivalentTo(vmPodIpAddress))
 
 				By("Deleting the VM.")
-				Expect(virtClient.VirtualMachine(vm.Namespace).Delete(vm.Name, &k8smetav1.DeleteOptions{})).To(Succeed())
-				tests.WaitForVirtualMachineToDisappearWithTimeout(vmi, 120)
+				Expect(cleanupVM(virtClient, vm.GetNamespace(), vm.GetName())).To(Succeed(), "should have been able to delete the VM object")
+				// this way, the test does not attempt to stop the VM in the `AfterEach` section
+				defer func() { vm = nil }()
```

Comment:
better if we just make the code at `AfterEach` to discover if it has to stop it or not.

Reply:
what's the advantage ?

Reply:
> what's the advantage ? It costs an extra read just to know if it must get rid of it.

Reply:
This is implementing cleanup code at an It

Reply:
> This is implementing cleanup code at an It

This is also what i suggested, in one of the threads, combine to removals, put them just on the AfterEach.
We can indeed try to delete the vm, if the delete has an error which isnt isnot found - assert, 
if its is not found - all good, if its found - wait for deletion of vmi, vm as you do now.

Reply:
> This is implementing cleanup code at an It

In case it doesnt hurt the test logic, 
this is also what i voted in one of the threads, combine the removals.
You can do something like
if vm != nil (you must, else you might deref nil),
delete - if error == isnotfound - all good
if error != nil , != isnotfound - assert
else wait for all gone

Reply:
> This is implementing cleanup code at an It

If this is cleanup code, what about removing the VM ? That's also cleanup code. But it's part of the test. 

My point is that those 2 things (deleting object / re-setting pointer) belong together.

Reply:
ok, fair enough